osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/providers/RsaEncryptionProvider.java

```diff
@@ -0,0 +1,67 @@
+package org.opensmartgridplatform.shared.security.providers;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.AlgorithmParameterSpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+
+import org.opensmartgridplatform.shared.exceptionhandling.EncrypterException;
+import org.opensmartgridplatform.shared.security.EncryptionProviderType;
+
+public class RsaEncryptionProvider extends AbstractEncryptionProvider implements EncryptionProvider {
+
+    private static final String ALG = "RSA";
+    private static final String ALGORITHM = "RSA/ECB/PKCS1Padding";
+
+    private Key publicKey;
+    private Key privateKey;
+
+    public void setPrivateKeyStore(File privateKeyStoreFile)
+            throws IOException, NoSuchAlgorithmException, InvalidKeySpecException {
+        byte[] keyData = Files.readAllBytes(privateKeyStoreFile.toPath());
+        PKCS8EncodedKeySpec privateKeySpec = new PKCS8EncodedKeySpec(keyData);
+        privateKey = KeyFactory.getInstance(ALG).generatePrivate(privateKeySpec);
+        super.setKeyFile(privateKeyStoreFile);
+    }
+
+    public void setPublicKeyStore(File publicKeyStoreFile) throws EncrypterException {
+        try {
+            byte[] keyData = Files.readAllBytes(publicKeyStoreFile.toPath());
+            X509EncodedKeySpec publicKeySpec = new X509EncodedKeySpec(keyData);
+            publicKey = KeyFactory.getInstance(ALG).generatePublic(publicKeySpec);
+        }
+        catch (NoSuchAlgorithmException | IOException | InvalidKeySpecException e) {
+            throw new EncrypterException("Could not set public keystore", e);
+        }
+    }
+
+    protected Cipher getCipher() throws EncrypterException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.opensmartgridplatform.shared.exceptionhandling.EncrypterException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)